TSPL.docc/ReferenceManual/Declarations.md

```diff
@@ -980,6 +1044,117 @@ see <doc:Functions#In-Out-Parameters>.
   ```
 -->
 
+#### Borrowing and Consuming Parameters
+
+The Swift compiler has varying ownership rules that it
+can use to manage object lifetimes across function calls.
+The default rules work well for most cases,
+but if necessary, you can use one of the `borrowing`
+or `consuming` modifiers to choose a specific ownership
+rule for a particular argument.
+
+Unlike `inout`, neither `borrowing` nor
+`consuming` parameters require any special
+notation when you call the function:
+
+```swift
+func someFunction(a: borrowing A, b: consuming B) { ... }
+
+someFunction(a: someA, b: someB)
+```
+
+The `borrowing` modifier indicates that the function
+does not expect to keep the value.
+In this case, the caller will retain ownership
+and ensure that the
+object remains alive,
+which minimizes overhead when the function
+is making only transient use of the object.
+However, if the function does try to keep the
+value - for example, by storing it in a global
+variable - the function will have to copy
+the value.
+
+```swift
+// `isLessThan` does not keep either argument
+func isLessThan(lhs: borrowing A, rhs: borrowing A) -> Bool {
+    ...
+}
+```
+
+Conversely, the `consuming` modifier indicates
+that the function *does* expect to keep the value,
+either to store it or to destroy it.
+This implicitly transfers ownership from the
+caller to the function,
+minimizing overhead when the caller no longer
+needs to use the object after the call.
+But if the caller does need to use the object
+after the call, it might need to make a private
+copy before calling the function.
+
+```swift
+// `store` keeps its argument, so we mark it `consuming`
+func store(a: consuming A) {
+    someGlobalVariable = a
+}
+```
+
+Note that Swift guarantees that object lifetime and
+ownership will be correctly managed in all cases,
+with or without one of these modifiers.
+These modifiers only impact the relative efficiency
+for particular patterns of usage.
```

Comment:
Let's move into the section intro — the reason to use `borrowing` or `consuming` is for efficiency (decrease the need for copying); the code is still correct without them.

Reply:
Done.